docs/content/contributing/Logging.md

```diff
@@ -46,3 +46,34 @@ val logger = loggerFactory.create("MyCustomTag")
 `StubLoggerFactory` and `StubLogger` can be used in tests
 
 `StubLogger` will write to stdout only during test runs from IDE
+
+## Verbose mode
+
+Use gradle property `avito.logging.verbosity` to override gradle logging level and send avito logs to stdout (e.g. with
+gradle's `quiet` level.)
+
+Value defines which levels to override.
+
+For example: `-Pavito.logging.verbosity=INFO` makes `INFO` and higher(`WARNING`) levels act like level quiet
+
+`CRITICAL`, which is mapped to gradle's `error` level is visible already on quiet level
+
+Possible values are in `DEBUG`, `INFO`, `WARNING`, `CRITICAL` (see `com.avito.logger.LogLevel`)
+
+Default is not defined
+
+### Stacktrace
+
+Add gradle's `--stacktrace` to also print stacktraces in verbose mode if available
+
+### Why is it needed?
+
+Gradle use lifecycle level by default, but to see info or debug level you have to set it for whole gradle run
+via `--info` or `--debug`, which made console output unreadable and build slow.
+
+There is an issue for
+that: [gradle/#1010 Ability to set log level for specific task](https://github.com/gradle/gradle/issues/1010)
```

Comment:
Maybe this will help:

```java
TaskInternal {
    /**
     * Sets the log message rewriter for this task's logger.
     */
    void setLoggerMessageRewriter(ContextAwareTaskLogger.MessageRewriter messageRewriter)
    
    interface MessageRewriter {

        /**
         * Rewrites log message.
         *
         * @param logLevel the logging level
         * @param message the original message
         * @return the rewritten message or null if this message should be silenced
         */
        @Nullable
        String rewrite(LogLevel logLevel, String message);
    }
    ```

Reply:
Agree, just wanted to consider alternative ways.